### DISCUSSION TITLE
commitMessageTopic setting ignored

### DISCUSSION THREAD

**Discussion by makarov-roman** [renovatebot/renovate#24278] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.87

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

Placeholder value, please select the correct response from the dropdown

### Describe the problem

Hi, I have a quite big monorepo with infra for eks clusters and I want to group dependency checks based on specific cluster.
I achieved grouping by using `"branchTopic": "{{{parentDir}}}"`, but I cannot update commitMessage/PRTitle to make it usable. This part is completely ignored.
```json5
 "commitMessageTopic": "cluster {{{parentDir}}}",
 "commitMessageExtra": "",
```
From what I managed to google is that sometimes related definitions for a datasource can overwrite provided global value. Which sounds like a bug, user-provided setting should always have more prio than defaults.
the dependency is terraform module located in private gitlab repo.
result title:
```json5
 "prTitle": "Update Terraform <url-to-module>",
```

Thanks in advance.

<details>
  <summary>Full renovate.rc</summary>

  ```json5
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    ":dependencyDashboard",
    ":label(dependencies)",
    ":enableVulnerabilityAlertsWithLabel(CVE)",
    ":prConcurrentLimit10",
    ":semanticCommitsDisabled"
  ],
  "rebaseWhen": "conflicted"
  "recreateWhen": "always",
  "assigneesFromCodeOwners": true,
  "cloneSubmodules": true,
  // group PR's by cluster
  "branchTopic": "{{{parentDir}}}",
  "commitMessageTopic": "cluster {{{parentDir}}}",
  "commitMessageExtra": ""
}
  ```  
</details>

I will create a minimal reproduction a bit later if this behaviour is not expected.

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment, marked as answer]
Try applying the settings using a package rule, Eg with matchDatasources plus the settings you need

**Reply by makarov-roman** [reply]
Thanks for the quick answer. Yes, it works. But isn't it more like workaround?

**Reply by viceice** [reply]
no, this is how it works.

**Reply by paddyroddy** [reply]
I would like to have `"commitMessageTopic": "{{depName}}"` always. Does that mean I would have to do add to the `matchManagers` list for every type I'm using?
